src/lib/TextDecorator/schemas.ts

```diff
@@ -1,8 +1,7 @@
 /* eslint import/prefer-default-export: 0 */
 
-import { ENS_DOMAIN_REGEX } from '../../modules/validations';
-
 const USER_PREFIX = '@';
+const ENS_DOMAIN_REGEX = '^([A-Za-z0-9](?:[A-Za-z0-9-.]{0,255}[A-Za-z0-9])?)';
```

Comment:
Could this go in the `validations.ts` file also, but with a different name? Probably also with comments about what the difference is between them - I am confused what this one is for! :)

Reply:
I really can't tell what this is all supposed to do, especially with the capture groups and the negated inner capture group.

As far as I can tell it needs to validate multiple occurrences of a ens name in a string _(as opposed to my version, which just validates the first occurrence)_

As far as I remember @chmanie wrote this, so maybe he has more details on it ?